modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -1452,6 +1452,13 @@ private static String getPrimitiveType(Schema schema) {
         } else if (ModelUtils.isUUIDSchema(schema)) {
             return "UUID";
         } else if (ModelUtils.isStringSchema(schema)) {
+            if(typeMapping.containsKey(schema.getFormat())) {
+                // If the format matches a typeMapping (supplied with the --typeMappings flag)
+                // then treat the format as a primitive type.
+                // This allows the typeMapping flag to add a new custom type which can then
+                // be used in the format field.   
+                return schema.getFormat();
```

Comment:
@steco Previously there's a special case handled in the code above: https://github.com/OpenAPITools/openapi-generator/pull/2799/files#diff-7cb46fa53f89a458a7b7cb201d2214a8R1417 (type: string, format: number).

If we just return format, would it be an issue if `schema.getFormat()` happens to be one of the existing OAS type (e.g. number)?

Should we return something like `string_number` instead to make it distinguishable with the existing types? Would that work in your case?

Reply:
I don't see the special case for type: string, format: number.  I see the following specific formats for strings - "date", "date-time", "password", "byte", "binary", "uuid", email" (in SchemaTypeUtil).

If someone has a swagger.json file which contains fields which are also primitive types, ie also in typeMapping (eg with type:string, format:number) then this change will mean that the field would change from being a string to being a double in the generated code.  However, I think this would be unlikely.

Using string_number would work for me, but I think this adds unnecessary complexity and makes a custom field different to the existing special cases (listed above).

Reply:
format could be anything. I don't think we should try to cover edge cases, as I think that would over-complicate things.

Reply:
I realize I'm not a contributor or member here, so I may not fully grasp everything in play here. However, I am a user who also needs this feature, and I implemented a very similar patch independently, so here's my opinion, for whatever it's worth.

@wing328's concern above appears to be that the edited code no longer handles a special case: string / number. However, that case still appears to be handled on (what is currently) line 1417:

    // Line 1417
    } else if (ModelUtils.isStringSchema(schema) && "number".equals(schema.getFormat())) {
        // special handle of type: string, format: number
        return "BigDecimal";
    } else if (ModelUtils.isByteArraySchema(schema)) {
    
Because that code is still there, it looks (to me, at least) like we can be confident it still handles the string/number case properly. Even better, that code didn't change and all edits to the function come after that test, which increases our confidence further.

I ran just this code change through the build and all tests pass, which is further evidence that this particular change does The Right Thing. (Note: In retrospect, I only ran surefire tests.)

With this change in, I have also confirmed that I can use type mappings to generate references to custom types instead of String for specific string/format combinations only.